tests/test_parameters.py

```diff
@@ -3,34 +3,61 @@
 
 
 class TestParameters(unittest.TestCase):
+    def setUp(self):
+        self.p = ModelParameters()
+        self.p.define("pA", 0.0)
+        self.p.define("pB", 0.0)
+
     def test_parameter(self):
-        p = ModelParameters()
-        p.define("pA", 0.0)
-        p.define("pB", 0.0)
-        p.update(["pA"], [42.0])
-        self.assertEqual(p["pA"], 42.0)
-        self.assertEqual(p["pB"], 0.0)
+        self.p.update(["pA"], [42.0])
+        self.assertEqual(self.p["pA"], 42.0)
+        self.assertEqual(self.p["pB"], 0.0)
+
+    def test_concat(self):
+        p1 = ModelParameters()
+        p1.define("A", 17)
+
+        p1 += self.p
+        self.assertEqual(p1["A"], 17)
+        self.assertEqual(p1["pA"], 0.0)
+        self.assertEqual(p1["pB"], 0.0)
+
+    def test_has(self):
+        self.assertTrue(self.p.has("pA"))
+        self.assertFalse(self.p.has("pC"))
+
+    def test_copy(self):
+        import copy
+
+        self.assertEqual(self.p["pA"], 0.0)
+        copy_p = copy.deepcopy(self.p)
+        copy_p["pA"] = 42.0
+        self.assertEqual(copy_p["pA"], 42)
+        self.assertEqual(self.p["pA"], 0.0)
+
+    def test_define(self):
+        self.assertRaises(Exception, self.p.__setitem__, "new_key", 0.2)
 
 
 class TestSingleModel(unittest.TestCase):
     def setUp(self):
         self.p = ModelParameters()
         self.p.define("pA", 0.0)
         self.p.define("pB", 0.0)
+        self.l = JointLatent()
+        self.l.add_model_parameters(self.p)
 
     def test_joint_list(self):
-        l = JointParameterList(self.p)
-        l.set_latent("pA")
-        l.update([42])
+        self.l.add("pA")
+        updated = self.l.update([42], return_copy=False)
         self.assertEqual(self.p["pA"], 42.0)
         self.assertEqual(self.p["pB"], 0.0)
 
         # not allowed to set latent, if you use these prior classes
-        self.assertRaises(Exception, UncorrelatedNormalPrior, l)
+        self.assertRaises(Exception, UncorrelatedNormalPrior, self.l)
```

Comment:
Not sure, if this prior test should be here, I would rather move this to a separate test of the prior class.